orchagent/switchorch.cpp

```diff
@@ -472,24 +478,230 @@ void SwitchOrch::doAppSwitchTableTask(Consumer &consumer)
+    const auto &hfSet = hash.lag_hash.value;
+
+    std::vector<sai_int32_t> hfList;
+    std::transform(
+        hfSet.cbegin(), hfSet.cend(), std::back_inserter(hfList),
+        [](sai_native_hash_field_t value) { return static_cast<sai_int32_t>(value); }
+    );
+
+    return this->setSwitchHashFieldListSai(oid, hfList);
+}
+
+bool SwitchOrch::setSwitchHash(const SwitchHash &hash)
 {
     SWSS_LOG_ENTER();
-    const string & table_name = consumer.getTableName();
 
-    if (table_name == APP_SWITCH_TABLE_NAME)
+    auto hObj = this->swHlpr.getSwHash();
+    auto cfgUpd = false;
+
+    if (hash.ecmp_hash.is_set)
+    {
+        if (hObj.ecmp_hash.value != hash.ecmp_hash.value)
+        {
+            if (this->swCap.isSwitchEcmpHashSupported())
+            {
+                if (!this->swCap.validateSwitchHashFieldCap(hash.ecmp_hash.value))
+                {
+                    SWSS_LOG_ERROR("Failed to validate switch ECMP hash: capability is not supported");
+                    return false;
+                }
+
+                if (!this->setSwitchHashEcmpHash(hash))
+                {
+                    SWSS_LOG_ERROR("Failed to set switch ECMP hash in SAI");
+                    return false;
+                }
+
+                cfgUpd = true;
+            }
+            else
+            {
+                SWSS_LOG_WARN("Switch ECMP hash configuration is not supported: skipping ...");
+            }
+        }
+    }
+    else
     {
-        doAppSwitchTableTask(consumer);
+        if (hObj.ecmp_hash.is_set)
+        {
+            SWSS_LOG_ERROR("Failed to remove switch ECMP hash configuration: operation is not supported");
+            return false;
+        }
     }
-    else if (table_name == CFG_ASIC_SENSORS_TABLE_NAME)
+
+    if (hash.lag_hash.is_set)
     {
-        doCfgSensorsTableTask(consumer);
+        if (hObj.lag_hash.value != hash.lag_hash.value)
+        {
+            if (this->swCap.isSwitchLagHashSupported())
+            {
+                if (!this->swCap.validateSwitchHashFieldCap(hash.lag_hash.value))
+                {
+                    SWSS_LOG_ERROR("Failed to validate switch LAG hash: capability is not supported");
+                    return false;
+                }
+
+                if (!this->setSwitchHashLagHash(hash))
+                {
+                    SWSS_LOG_ERROR("Failed to set switch LAG hash in SAI");
+                    return false;
+                }
+
+                cfgUpd = true;
+            }
+            else
+            {
+                SWSS_LOG_WARN("Switch LAG hash configuration is not supported: skipping ...");
+            }
+        }
     }
     else
     {
-        SWSS_LOG_ERROR("Unknown table : %s", table_name.c_str());
+        if (hObj.lag_hash.is_set)
+        {
+            SWSS_LOG_ERROR("Failed to remove switch LAG hash configuration: operation is not supported");
+            return false;
+        }
     }
 
+    if (!cfgUpd)
```

Comment:
Please add a comment here. Not clear on why its returning early

Reply:
@prsunny done